jelatyna-backend/src/main/java/pl/confitura/jelatyna/login/OAuthUserService.java

```diff
@@ -1,23 +1,22 @@
 package pl.confitura.jelatyna.login;
 
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
-import pl.confitura.jelatyna.user.User;
-import pl.confitura.jelatyna.user.UserRepository;
+import pl.confitura.jelatyna.user.dto.User;
+import pl.confitura.jelatyna.user.UserFacade;
 
+@RequiredArgsConstructor
 @Service
 public class OAuthUserService {
-    private UserRepository userRepository;
 
-    public OAuthUserService(UserRepository userRepository) {
-        this.userRepository = userRepository;
-    }
+    private final UserFacade userFacade;
 
     public User mapToUser(OAuthUserBase oauthUser) {
         String id = oauthUser.encodeId();
-        if (!userRepository.existsBySocialId(id)) {
-            userRepository.save(oauthUser.toUser());
+        if (!userFacade.existsUserWithSocialId(id)) {
```

Comment:
troche mi ta nazwa trzeszczy... Dla mnie bardziej po angielsku bylo by `userWithSocialIdExists` albo nawet cos w stylu `userExists(SocialId id)` ale to by wymagalo klas na IDki, co jest do przemyslania, ale nie koniecznie w tym PRe.